tee-worker/scripts/litentry/release/deploy.sh

```diff
@@ -1,736 +1,555 @@
+  echo "Stopping worker services ..."
+  for ((i = 0; i < WORKER_COUNT; i++)); do
+    sudo systemctl stop "worker$i.service"
+    sleep 5
+  done
+}
 
-  cp ./${service_name}.service /etc/systemd/system/
-  systemctl daemon-reload
-  systemctl start $service_name
+function stop_parachain_services {
+  echo "Stopping parachain services ..."
+  sudo systemctl stop para-alice.service relay-alice.service relay-bob.service
+}
 
-  sleep 10
+function stop_services {
+  stop_worker_services
 
-  echo "Finished restarting Parachain, Check logs at /tmp/parachain_dev/para.alice.log"
+  # TODO: it means we can't stop parachain service alone
+  #       this needs to be done directly via `systemctl`
+  if [ "$ONLY_WORKER" = false ]; then
+    stop_parachain_services
+  fi
 }
 
-function register_parachain() {
-  echo "register parathread now ..."
-  cd "$ROOTDIR/ts-tests" || exit
-  if [[ -z "${NODE_ENV}" ]]; then
+function register_parachain {
+  echo "Register parathread now ..."
+  cd "$ROOTDIR" || exit
+  export PARACHAIN_ID=$(grep DEFAULT_PARA_ID node/src/chain_specs/$CHAIN.rs  | grep u32 | sed 's/.* = //;s/\;//')
+  cd "$ROOTDIR/ts-tests" || exit 
+  if [[ -z "$NODE_ENV" ]]; then
       echo "NODE_ENV=ci" > .env
   else
-      echo "NODE_ENV=${NODE_ENV}" > .env
+      echo "NODE_ENV=$NODE_ENV" > .env
   fi
-  # The genesis state path file needs to be updated as it is hardcoded to be /tmp/parachain_dev
-  jq --arg genesis_state "$TMPDIR/genesis-state" --arg genesis_wasm "$TMPDIR/genesis-wasm" '.genesis_state_path = $genesis_state | .genesis_wasm_path = $genesis_wasm' config.ci.json > updated_config.json
-  mv updated_config.json config.ci.json
+  # The genesis state path file needs to be updated as it is hardcoded to be /tmp/parachain_dev 
+  jq --arg genesis_state "$PARACHAIN_BASEDIR/genesis-state" --arg genesis_wasm "$PARACHAIN_BASEDIR/genesis-wasm" '.genesis_state_path = $genesis_state | .genesis_wasm_path = $genesis_wasm' config.ci.json > config.ci.json.1
+  mv config.ci.json.1 config.ci.json
   corepack yarn
-  corepack yarn register-parathread 2>&1 | tee "$TMPDIR/register-parathread.log"
+  corepack yarn register-parathread 2>&1 | tee "$PARACHAIN_BASEDIR/register-parathread.log"
   print_divider
 
-  echo "upgrade parathread to parachain now ..."
+  echo "Upgrade parathread to parachain now ..."
   # Wait for 90s to allow onboarding finish, after that we do the upgrade
   sleep 90
-  cd "$ROOTDIR/ts-tests" || exit
-  if [[ -z "${NODE_ENV}" ]]; then
-      echo "NODE_ENV=ci" > .env
-  else
-      echo "NODE_ENV=${NODE_ENV}" > .env
-  fi
-  corepack yarn
-  corepack yarn upgrade-parathread 2>&1 | tee "$TMPDIR/upgrade-parathread.log"
+  corepack yarn upgrade-parathread 2>&1 | tee "$PARACHAIN_BASEDIR/upgrade-parathread.log"
   print_divider
 
-  echo "done. please check $TMPDIR for generated files if need"
-
+  echo "done. please check $PARACHAIN_BASEDIR for generated files if need"
   print_divider
+  git restore config.ci.json
```

Comment:
Doesn't restore remove any changes we made to the `config.ci.json` in the earlier step with `jq`.

Reply:
Yea but we need it just for the parathread upgrade, don't we? And we should be done with it here